### ISSUE TITLE
Weird issue seen once in DeferredJVM tests

### ISSUE THREAD

**Issue by djspiewak** [typelevel/cats-effect#1008]
```
[info] DeferredJVMParallelism1Tests
Exception in thread "io-compute-5" java.lang.NullPointerException
  at cats.effect.ByteStack.checkAndGrow(ByteStack.scala:67)
  at cats.effect.ByteStack.push(ByteStack.scala:27)
  at cats.effect.IOFiber.cats$effect$IOFiber$$pushCont(IOFiber.scala:607)
  at cats.effect.IOFiber.run(IOFiber.scala:205)
  | => cat cats.effect.IOFiber.$anonfun$8$$anonfun$4$$anonfun$4(IOFiber.scala:534)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  at java.lang.Thread.run(Thread.java:748)
```

Only saw it once on an orphaned branch. Recording for posterity.

**Comment by djspiewak** [comment]
Related: 

```
Exception in thread "io-compute-cats.effect.concurrent.DeferredSpec-4" java.lang.NullPointerException
	at cats.effect.CallbackStack.apply(CallbackStack.scala:44)
	at cats.effect.IOFiber.cats$effect$IOFiber$$done(IOFiber.scala:225)
	at cats.effect.IOFiber.cats$effect$IOFiber$$asyncCancel(IOFiber.scala:294)
	at cats.effect.IOFiber.cats$effect$IOFiber$$runLoop(IOFiber.scala:313)
	at cats.effect.IOFiber.run(IOFiber.scala:212)
	at cats.effect.IOFiber.$anonfun$runLoop$8(IOFiber.scala:537)
  | => cat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Also… twice!

**Comment by RaasAhsan** [comment]
I think both of these are solved with #1012 and #1014
